src/argilla/client/feedback/metrics/annotator_metrics.py

```diff
@@ -0,0 +1,373 @@
+            kwargs: additional arguments to pass to the metric.
+
+        Raises:
+            ValueError: If the metric name is not supported for the given question.
+
+        Returns:
+            metrics: dict with the metrics computed for each annotator, where the
+                key corresponds to the user id and the values are a list with the
+                metric results.
+        """
+        if isinstance(metric_names, str):
+            metric_names = [metric_names]
+
+        if any([metric not in self._allowed_metrics for metric in metric_names]):
+            raise ValueError(
+                f"Metrics allowed for question {self._question_name}: {list(self._allowed_metrics.keys())}"
+            )
+
+        metric_classes = [(metric_name, self._allowed_metrics[metric_name]) for metric_name in metric_names]
+
+        responses_per_user, suggestions = get_responses_per_user(self._dataset, self._question_name)
+
+        metrics = defaultdict(list)
+        for user_id, responses in responses_per_user.items():
+            for metric_name, metric_cls in metric_classes:
+                metric = metric_cls(responses=responses, suggestions=suggestions)
+                result = metric.compute(**kwargs)
+                metrics[user_id].append(AnnotatorMetricResult(metric_name=metric_name, result=result))
+
+        return dict(metrics)
+
+
+class AnnotatorMetricBase(ABC):
+    """Base class for Annotator metrics."""
+
+    def __init__(self, responses=None, suggestions=None) -> None:
+        """
+        Args:
+            responses: Responses given by the user.
+                Depending on the type of question it can be a list of strings, or integers.
+            suggestions: Suggestions offered for the annotators.
+                Same format as `responses`.
+        """
+        self._responses = responses
+        self._suggestions = suggestions
+
+    def compute(self, **kwargs):
+        responses, suggestions = self._pre_process(self._responses, self._suggestions)
+        return self._compute(responses, suggestions, **kwargs)
+
+    def _pre_process(self, responses, suggestions) -> Any:
+        """Optional data preprocessing. By default it just passes the data to the _compute method.
+
+        Args:
+            responses: Responses given by the user.
+            suggestions: Suggestions offered for the annotators.
+
+        Returns:
+            data: tuple with the preprocessed data.
+        """
+        return responses, suggestions
+
+    @abstractmethod
+    def _compute(self, responses, suggestions, **kwargs):
+        """Abstract method where the computation is done.
+
+        Args:
+            responses: Responses given by the user, as expected by the given metric.
+            suggestions: Suggestions offered for the annotators, as expected by the given metric.
+        """
+        pass
+
+
+def is_multiclass(data) -> bool:
```

Comment:
maybe this might go into helper/utils directory to adhere with the rest of the project structure?